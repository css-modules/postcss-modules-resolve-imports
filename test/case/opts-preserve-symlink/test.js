'use strict';

const resolveImports = require('../../../index');
const setup = require('../../setup');

test('opts-preserve-symlinks true', () => {
  const {resulting, exports: tokens} = setup(
    'local-by-default',
    'extract-imports',
    'scope',
    resolveImports({
      resolve: {
        preserveSymlinks: true,
      },
    })
  )(__dirname);

  expect(resulting).toMatchSnapshot();
  expect(tokens).toMatchSnapshot();
});

test('opts-preserve-symlinks false', () => {
  const {resulting, exports: tokens} = setup(
    'local-by-default',
    'extract-imports',
    'scope',
    resolveImports({
      resolve: {
        preserveSymlinks: false,
      },
    })
  )(__dirname);

  expect(resulting).toMatchSnapshot();
  expect(tokens).toMatchSnapshot();
});

test('opts-preserve-symlinks default false', () => {
  const {resulting, exports: tokens} = setup(
    'local-by-default',
    'extract-imports',
    'scope',
    // nodejs behaviour: defaults to false
    resolveImports({})
  )(__dirname);

  expect(resulting).toMatchSnapshot();
  expect(tokens).toMatchSnapshot();
});
