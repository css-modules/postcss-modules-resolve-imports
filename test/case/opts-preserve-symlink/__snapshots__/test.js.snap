// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opts-preserve-symlinks default false 1`] = `
"/*  imported from lib/button/index.css  */

._index_button {
  color: green;
}

/*  imported from source.css  */

._source_continueButton
{
  color: green;
}
"
`;

exports[`opts-preserve-symlinks default false 2`] = `
Object {
  "continueButton": "_source_continueButton _index_button",
}
`;

exports[`opts-preserve-symlinks false 1`] = `
"/*  imported from lib/button/index.css  */

._index_button {
  color: green;
}

/*  imported from source.css  */

._source_continueButton
{
  color: green;
}
"
`;

exports[`opts-preserve-symlinks false 2`] = `
Object {
  "continueButton": "_source_continueButton _index_button",
}
`;

exports[`opts-preserve-symlinks true 1`] = `
"/*  imported from node_modules/button/index.css  */

._index_button {
  color: green;
}

/*  imported from source.css  */

._source_continueButton
{
  color: green;
}
"
`;

exports[`opts-preserve-symlinks true 2`] = `
Object {
  "continueButton": "_source_continueButton _index_button",
}
`;
